services/src/main/java/io/druid/cli/validate/DruidJsonValidator.java

```diff
@@ -69,4 +110,40 @@ public void run()
       throw Throwables.propagate(e);
     }
   }
+
+  private List<String> readData(final StringInputRowParser parser, final CharSource source, final boolean collect)
+      throws IOException
+  {
+    return source.readLines(
+        new LineProcessor<List<String>>()
+        {
+          private final List<String> collected = new ArrayList<String>();
+          private final StringBuilder builder = new StringBuilder();
+
+          @Override
+          public boolean processLine(String line) throws IOException
+          {
+            InputRow parsed = parser.parse(line);
+            builder.append(parsed.getTimestamp());
+            for (String dimension : parsed.getDimensions()) {
+              builder.append('\t');
+              builder.append(parsed.getRaw(dimension));
+            }
+            if (collect) {
+              collected.add(builder.toString());
+            } else {
+              System.out.println(builder.toString());
```

Comment:
I thought the output would be redirected to a log file unexpectedly. I'll add an option for that.